opentracing-api/src/main/java/io/opentracing/ContinuableScope.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2016-2017 The OpenTracing Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package io.opentracing;
+
+/**
+ * A {@link ContinuableScope} formalizes the deferral of state of a {@link Span} for cross-thread scenarios.
+ *
+ * <p>
+ * When doing cross-thread tracing, a {@link Span} state may need to be deferred and later recovered on a
+ * different thread. This interface formalizes this advanced scenario, by extending {@link Scope} and
+ * adding {@link ContinuableScope#defer()} and exposing {@link Continuation} used to reactivate
+ * the {@link Span}.
+ */
+public interface ContinuableScope extends Scope {
```

Comment:
could it be implemented as a wrapper around the scope instead? Given my other comment about refcounting class below, I could see these uses:
```java

try (Scope scope = tracer.buildSpan("parent").startActive(Scope.FINISH_ON_CLOSE)) {
  // cannot defer because span will finish on exit
}

try (ContinuableScope scope = ContinuableScope.of(tracer.buildSpan("parent").startActive())) {
  // must defer because otherwise span won't finish
}

try (ContinuableScope scope = RefCountedScope.of(tracer.buildSpan("parent").startActive(???))) {
  // RefCountedScope extends ContinuableScope
  // code may or may not defer, span will finish correctly either way
}

```

Reply:
```
try (ContinuableScope scope = ContinuableScope.of(tracer.buildSpan("parent").startActive())) {
```
This wouldn't need to defer - a `ContinuableScope` is simply a `Scope` with `defer()` support :)

Otherwise, yes, this could as well work fine, as it won't be touching the main API, and having this as a class, instead of an interface - but I'm wondering how comfortable would people be having such (at least the `Continuable` portion) in `opentracing-util`, if given the case?